### DISCUSSION TITLE
Github Oauth plugin for jenkins

### DISCUSSION THREAD

**Discussion by shevam9** [community/community#121888] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have set up github authentication in my jenkins by setting up a Oauth app in my github, but it seems that now anyone with a github account can login into my jenkins. How to restrict that to a specific no of users. 

Also it is not about authorization as i don't want them logging into jenkins itself in the first place, access after that is another matter.

**Comment by markolalovic** [comment]
How did you set up the authorization strategy? In the documentation: [Jenkins GitHub OAuth Plugin](https://plugins.jenkins.io/github-oauth/#plugin-content-automatically-configure-authorization-strategy-via-script-console), you can find an explanation of how to configure different strategies.

**Reply by shevam9** [reply]
i understand that by configuring authorization i can controll access but still any github user can still enter my jenkins controller even though they don't have any read or overall access. My doubt is that entry point still poses some security threats right?

**Reply by markolalovic** [reply]
What credentials are you using to perform connection to your Jenkins controller? "No verification" host key verification strategy is not recommended.

**Reply by shevam9** [reply]
My fellow team members are logging into the controller via jenkins UI, they are using their github credentials , they get a authorization prompt and code in their mail , so upon verifying that they are able to login into the controller. There is no such option for host key verification strategy in regards to the github Oauth plugin.

**Reply by markolalovic** [reply]
You wrote: "it seems that now anyone with a github account can login into my jenkins". Why does it seem like this, have you tested with fellow members?

**Reply by shevam9** [reply]
Yes anyone is able to log in to the controller,  i tried with multiple github accounts, i can only restrict the inside access like read or overall read etc based on what authorization policy i am using but logging in is possible. I am actually looking for a way i can restrict login to specific accounts but nothing so far.

**Reply by markolalovic** [reply]
It seems like you have granted overall/administrative permission to anonymous (anyone can do anything). What you want is granting that permission to authenticated users. Perhaps check this related question and answer at [server fault](https://serverfault.com/questions/789420/jenkins-login-using-github-authentication-failed-being-administrator) to try and debug your permissions.
